internal/envelope/ipld.go

```diff
@@ -118,37 +135,53 @@ func FromDagJsonReader[T Tokener](r io.Reader) (T, cid.Cid, error) {
 func FromIPLD[T Tokener](node datamodel.Node) (T, cid.Cid, error) {
 	undef := *new(T)
 
-	signatureNode, err := node.LookupByIndex(0)
+	id, err := CIDFromIPLD(node)
```

Comment:
there is still paths where this (additional encoding) is not necessary. Sometimes it comes where a spot where it's already cbor being decoded